src/utilities/parse-pulp-id.ts

```diff
@@ -10,3 +12,195 @@ export function parsePulpIDFromURL(url: string): string {
 
   return null;
 }
+
+export function parsePulpIDFromPRN(prn: string): string {
+  // PRNs are of the form prn:app_label.model_label:pulp_id
+  const sections = prn.split(':');
+  if (sections.length == 3 && sections[2].match(UUIDRegEx)) {
+    return sections[2];
+  }
+
+  return null;
+}
+
+interface PulpResource {
+  id: string;
+  plugin: string;
+  model: string;
+  resource_label?: string;
+}
+
+export function parsePulpResource(value: string): PulpResource {
+  // value can be a url or a prn
+  let id = '',
+    plugin = '',
+    model = '',
+    resource_label;
+  if (value.startsWith(API_BASE_PATH)) {
+    const url = value.replace(API_BASE_PATH, '');
+    id = parsePulpIDFromPRN(url);
+    [plugin, model] = translateURLToModelLabel(url).split('.');
+  } else if (value.startsWith('prn:')) {
+    // prn is of the form `prn:app_label.model_label:pulp_id`
+    let label;
+    [, label, id] = value.split(':');
+    [plugin, model] = label.split('.');
+  } else if (value.startsWith('pdrn:')) {
+    // pdrn is of the form `pdrn:domain_pulp_id:resource_label`
+    [, id, resource_label] = value.split(':');
+    [plugin, model] = ['core', 'domain'];
+  } else {
+    return null;
+  }
+  return {
+    id: id,
+    plugin: plugin,
+    model: model,
+    resource_label: resource_label,
+  };
+}
+
+export function translatePulpResourceToURL(resource: PulpResource): string {
+  // Doesn't handle PulpResources from pdrns
+  const model_label = resource.plugin + '.' + resource.model;
+  const urlPrefix = ModelLabelToURL[model_label];
+  return API_BASE_PATH + urlPrefix + '/' + resource.id + '/';
+}
+
+function translateURLToModelLabel(url: string): string {
+  const value = url.replace(API_BASE_PATH, '');
+  const urlParts = value.split('/').filter((part) => !part.match(IDRegEx));
```

Comment:
@gerrod3 Good point. I had a better look and here's the tricky part: 

`IDRegEx` is `/^[0-9a-f-]+/i` so "domains", "ansible", "collections" get filtered out as well so they never make it at all to `URLToModelLabel`. Using `UUIDRegEx` seems to match better but it matches empty string as well so that needs to be filtered out later(`UUIDRegEx` is used via `parsePulpIDFromURL` at too many places so I'd prefer not to touch it if possible) or it's trying to match `domains\` instead of `domains`. So something like this:

```diff
diff --git a/src/utilities/parse-pulp-id.ts b/src/utilities/parse-pulp-id.ts
index d8768a1..8c34d7c 100644
--- a/src/utilities/parse-pulp-id.ts
+++ b/src/utilities/parse-pulp-id.ts
@@ -69,8 +69,8 @@ export function translatePulpResourceToURL(resource: PulpResource): string {
 
 function translateURLToModelLabel(url: string): string {
   const value = url.replace(API_BASE_PATH, '');
-  const urlParts = value.split('/').filter((part) => !part.match(IDRegEx));
-  const searchUrl = urlParts.join('/');
+  const urlParts = value.split('/').filter((part) => !part.match(UUIDRegEx));
+  const searchUrl = urlParts.filter((part) => part !== '').join('/');
   return URLToModelLabel[searchUrl];
 }
```

Reply:
Ok, I added `$` to prevent partial matches. The reason we can't use `UUIDRegEx` is because some urls have normal integer ids in them, e.g. user hrefs, repo-version hrefs.